go/coredns/plugin/pfdns/pfdns.go

```diff
@@ -636,9 +662,50 @@ func (pf *pfdns) SetPassthrough(ctx context.Context, passthrough, ip, port strin
 	return err
 }
 
-func (pf *pfdns) PortalFQDNInit() error {
+func (pf *pfdns) PortalFQDNInit(ctx context.Context) error {
 	general := pfconfigdriver.Config.PfConf.General
-	pf.PortalFQDN = general.Hostname + "." + general.Domain + "."
+
+	index := 0
+
+	pf.PortalFQDN = make(map[int]map[*net.IPNet]*regexp.Regexp)
+
+	var interfaces pfconfigdriver.ListenInts
+	pfconfigdriver.FetchDecodeSocket(ctx, &interfaces)
+
+	var keyConfNet pfconfigdriver.PfconfigKeys
+	keyConfNet.PfconfigNS = "config::Network"
+	keyConfNet.PfconfigHostnameOverlay = "yes"
+	pfconfigdriver.FetchDecodeSocket(ctx, &keyConfNet)
+
+	var NetIndex net.IPNet
+
+	for _, key := range keyConfNet.Keys {
+		var ConfNet pfconfigdriver.NetworkConf
+		ConfNet.PfconfigHashNS = key
+		pfconfigdriver.FetchDecodeSocket(ctx, &ConfNet)
+
+		var fqdn string
+		if ConfNet.PortalFQDN != "" {
+			fqdn = ConfNet.PortalFQDN
+		} else {
+			fqdn = general.Hostname + "." + general.Domain
+		}
+
+		NetIndex.Mask = net.IPMask(net.ParseIP(ConfNet.Netmask))
+		NetIndex.IP = net.ParseIP(key)
+
+		rgx, _ := regexp.Compile(".*" + fqdn)
+
+		pf.PortalFQDN[index] = make(map[*net.IPNet]*regexp.Regexp)
+		pf.PortalFQDN[index][&NetIndex] = rgx
```

Comment:
You are reusing the same NetIndex over and over again.
Within the loop do this 
 ```NetIndex  := &net.IPNet{}```
To create a new net.IPNet time instead of reusing the same address.